packages/firebase_admob/example/android/app/src/main/java/io/flutter/plugins/firebaseadmobexample/MainActivity.java

```diff
@@ -8,12 +8,16 @@
 import io.flutter.embedding.android.FlutterActivity;
 import io.flutter.embedding.engine.FlutterEngine;
 import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin;
+import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin.NativeAdFactory;
 
 public class MainActivity extends FlutterActivity {
   // TODO(bparrishMines): Remove this once v2 of GeneratedPluginRegistrant rolls to stable. https://github.com/flutter/flutter/issues/42694
   @Override
   public void configureFlutterEngine(FlutterEngine flutterEngine) {
     flutterEngine.getPlugins().add(new E2EPlugin());
     flutterEngine.getPlugins().add(new FirebaseAdMobPlugin());
+
+    final NativeAdFactory factory = new NativeAdFactoryExample(getLayoutInflater());
+    FirebaseAdMobPlugin.registerNativeAdFactory(flutterEngine, "adFactoryExample", factory);
```

Comment:
We should probably call unregisterNativeAdFactory on `cleanUpFlutterEngine`.

For this and the previous, we'd probably want to update the examples in README as well.

Reply:
Do we need to do that? I thought cleanup such as that isn't necessary for the V2 Embedding. (e.g. calling `MethodChannel.setMethodCallHandler(null)`). `FirebaseAdmobPlugin` doesn't maintain any static references, so the reference to the `NativeAdFactory` would be destroyed along with the `FirebaseAdmobPlugin` instance.